assets/dev/js/frontend/handlers/base-carousel.js

```diff
@@ -199,4 +269,66 @@ export default class CarouselHandlerBase extends SwiperHandlerBase {
 
 		this.swiper.update();
 	}
+
+	getPaginationBullets( array = true ) {
+		const paginationBullets = this.$element.find( this.getSettings( 'selectors' ).paginationBullet );
+
+		return array ? Array.from( paginationBullets ) : paginationBullets;
+	}
+
+	a11ySetWidgetAriaDetails() {
+		const $widget = this.$element;
+
+		$widget.attr( 'aria-roledescription', 'carousel' );
+		$widget.attr( 'aria-label', elementorFrontend.config.i18n.a11yCarouselWrapperAriaLabel );
+	}
+
+	a11ySetPaginationTabindex() {
+		this.getPaginationBullets().forEach( ( bullet ) => {
+			if ( ! bullet.classList.contains( 'swiper-pagination-bullet-active' ) ) {
+				bullet.removeAttribute( 'tabindex' );
+			}
+		} );
+
+		const isDirectionInlineArrowKey = 'ArrowLeft' === event?.code || 'ArrowRight' === event?.code;
+
+		if ( event?.target.classList.contains( 'swiper-pagination-bullet' ) && isDirectionInlineArrowKey ) {
+			this.$element.find( '.swiper-pagination-bullet-active' ).trigger( 'focus' );
+		}
+	}
+
+	getSwiperWrapperTranformXValue() {
+		let transformValue = this.elements.$swiperWrapper[ 0 ]?.style.transform;
+		transformValue = transformValue.replace( 'translate3d(', '' );
+		transformValue = transformValue.split( ',' );
+		transformValue = parseInt( transformValue[ 0 ].replace( 'px', '' ) );
+
+		return !! transformValue ? transformValue : 0;
+	}
+
+	a11ySetSlideAriaHidden( status = '' ) {
+		const currentIndex = 'initialisation' === status ? 0 : this.swiper?.activeIndex;
+
+		if ( 'number' !== typeof currentIndex ) {
+			return;
+		}
+
+		const swiperWrapperTransformXValue = this.getSwiperWrapperTranformXValue(),
+			swiperWrapperWidth = this.elements.$swiperWrapper[ 0 ].clientWidth,
+			$slides = this.elements.$swiperContainer.find( this.getSettings( 'selectors' ).slideContent );
+
+		$slides.each( ( index, slide ) => {
+			const isSlideInsideWrapper = 0 <= ( slide.offsetLeft + swiperWrapperTransformXValue ) && ( swiperWrapperWidth > ( slide.offsetLeft + swiperWrapperTransformXValue ) );
+
+			if ( ! isSlideInsideWrapper ) {
+				slide.setAttribute( 'aria-hidden', true );
+				slide.setAttribute( 'inert', '' );
+			} else {
+				slide.removeAttribute( 'aria-hidden' );
+				slide.removeAttribute( 'inert' );
+			}
+		} );
+	}
+
+	handleElementHandlers() {}
```

Comment:
@hein-obox Maybe just add a comment here that it is an empty method so that child classes can override this method.

Reply:
Updated.